internal/provisioner/cluster.go

```diff
@@ -100,32 +124,40 @@ func CreateCluster(provider, s3StateStore, size string, zones []string, wait int
 
 	if wait > 0 {
 		logger.Infof("waiting up to %d seconds for k8s cluster to become ready...", wait)
-		err = kops.WaitForKubernetesReadiness(dns, wait)
+		err = kops.WaitForKubernetesReadiness(kopsMetadata.Name, wait)
 		if err != nil {
 			// Run non-silent validate one more time to log final cluster state
 			// and return original timeout error.
-			kops.ValidateCluster(dns, false)
+			kops.ValidateCluster(kopsMetadata.Name, false)
 			return err
 		}
 	}
 
-	logger.WithField("dns", dns).Info("successfully created cluster")
+	logger.WithField("name", kopsMetadata.Name).Info("successfully created cluster")
 
 	return nil
 }
 
 // UpgradeCluster upgrades a cluster to the latest recommended production ready k8s version.
-func UpgradeCluster(clusterID, s3StateStore string, wait int, logger log.FieldLogger) error {
-	logger = logger.WithField("cluster", clusterID)
+func UpgradeCluster(cs clusterStore, clusterID, s3StateStore string, wait int, logger log.FieldLogger) error {
```

Comment:
Just checking: eventually we will want to store some extra data like k8s version in the DB, right?

Reply:
In all likelihood -- yes. Unclear if we want this in the top-level Clusters schema, or if it would be part of the provisioner metadata.